tests/queries/0_stateless/03277_json_subcolumns_in_primary_key.sql.j2

```diff
@@ -0,0 +1,71 @@
+-- Tags: long
+
+set optimize_read_in_order=1;
+set allow_experimental_json_type=1;
+set allow_experimental_variant_type=1;
```

Comment:
I see the variant type being enabled here, but I don't see it being used in the tests?

Reply:
It's needed to run this type of query:
```
insert into test select number, toJSONString(map('a', number % 2, 'b', 'str_' || number % 3, 'c', range(number % 10))), 'Hello, World!' from numbers(4);
```

By default it will not be able to create such map with an exception:
```
There is no supertype for types UInt8, String, Array(UInt8)
```

But with `use_variant_as_common_type=1` and `allow_experimental_variant_type=1` it will return Map with type `Map(String, Variant(Array(UInt8), String, UInt8))`. It's just easier to create a JSON with paths with different types this way